pkg/strategy/oneliner/strategy.go

```diff
@@ -0,0 +1,420 @@
+	var instanceID = s.InstanceID()
+
+	if s.Position == nil {
+		s.Position = types.NewPositionFromMarket(s.Market)
+	}
+
+	if s.ProfitStats == nil {
+		s.ProfitStats = types.NewProfitStats(s.Market)
+	}
+
+	if s.TradeStats == nil {
+		s.TradeStats = types.NewTradeStats(s.Symbol)
+	}
+
+	// StrategyController
+	s.Status = types.StrategyStatusRunning
+
+	s.OnSuspend(func() {
+		// Cancel active orders
+		_ = s.orderExecutor.GracefulCancel(ctx)
+	})
+
+	s.OnEmergencyStop(func() {
+		// Cancel active orders
+		_ = s.orderExecutor.GracefulCancel(ctx)
+		// Close 100% position
+		//_ = s.ClosePosition(ctx, fixedpoint.One)
+	})
+
+	// initial required information
+	s.session = session
+
+	s.orderExecutor = bbgo.NewGeneralOrderExecutor(session, s.Symbol, ID, instanceID, s.Position)
+	s.orderExecutor.BindEnvironment(s.Environment)
+	s.orderExecutor.BindProfitStats(s.ProfitStats)
+	s.orderExecutor.BindTradeStats(s.TradeStats)
+
+	//modify := func(p float64) float64 {
+	//	return p
+	//}
+	//if s.GraphPNLDeductFee {
+	//	modify = func(p float64) float64 {
+	//		return p * (1. - Fee)
+	//	}
+	//}
+	profit := floats.Slice{1., 1.}
+	price, _ := s.session.LastPrice(s.Symbol)
+	initAsset := s.CalcAssetValue(price).Float64()
+	cumProfit := floats.Slice{initAsset, initAsset}
+	s.orderExecutor.TradeCollector().OnTrade(func(trade types.Trade, _profit, netProfit fixedpoint.Value) {
+		profit.Update(netProfit.Float64())
+		cumProfit.Update(s.CalcAssetValue(trade.Price).Float64())
+	})
+	s.orderExecutor.TradeCollector().OnPositionUpdate(func(position *types.Position) {
+		bbgo.Sync(s)
+	})
+	s.orderExecutor.Bind()
+	s.activeOrders = bbgo.NewActiveOrderBook(s.Symbol)
+
+	for _, method := range s.ExitMethods {
+		method.Bind(session, s.orderExecutor)
+	}
+
+	store, _ := session.MarketDataStore(s.Symbol)
+	s.nrr = &NRR{IntervalWindow: types.IntervalWindow{Window: 2, Interval: s.Interval}, RankingWindow: s.Window}
+	s.nrr.Bind(store)
+
+	//startTime := s.Environment.StartTime()
+	//s.TradeStats.SetIntervalProfitCollector(types.NewIntervalProfitCollector(types.Interval1h, startTime))
+
+	// queued first signal as its initial process
+	//alphaLst := types.NewQueue(2)
+
+	s.session.MarketDataStream.OnKLineClosed(types.KLineWith(s.Symbol, s.Interval, func(kline types.KLine) {
+
+		// transformed to [0~1] which divided equally
+		//alpha := s.nrr.RankedValues.Last()
+
+		// alpha-weighted assets (inventory and capital)
+		targetBase := s.QuantityOrAmount.CalculateQuantity(kline.Close).Mul(fixedpoint.NewFromFloat(s.nrr.RankedValues.Index(1)))
+		diffQty := targetBase.Sub(s.Position.Base)
+
+		log.Infof("decision alpah: %f, ranked negative return: %f, current position: %f, target position diff: %f", s.nrr.RankedValues.Index(1), s.nrr.RankedValues.Last(), s.Position.Base.Float64(), diffQty.Float64())
+
+		// use kline direction to prevent reversing position too soon
+		if diffQty.Sign() > 0 { // && kline.Direction() >= 0
```

Comment:
if diffQty.Sign() > 0 && kline.Direction() >= 0 {

Reply:
Will check this, thanks for reporting.